### PR TITLE
refactor: remove use of payload files

### PR DESCRIPTION
**Pull request by browningjp-datatonic**
# Description

Refactor that removes the use of the payload JSON files. This is expected to significantly improve the ease-of-use of the codebase, and reduce the (duplicate) configuration required.

- Pipeline parameters for use in the dev/sandbox environment are defined in the pipeline definition itself (the default arguments for the pipeline function in `pipeline.py`). These values are also used during the E2E tests, including in CI/CD. Some of these values are taken from environment variables (using `os.environ.get(...)` so can be altered in the Cloud Build trigger
- Pipeline parameters for use in the test and prod environments (via Cloud Scheduler) are defined in the Terraform configuration for those environments
- The E2E tests no longer need to override the `pipeline_files_gcs_path` using an environment variable in `trigger/main.py`, since the GCS directory used for the assets files is defined as an environment variable anyway (using `os.environ.get` in the pipeline's default arguments)

# How has this been tested?

Unit tests all passing (including trigger-tests)

Full end-to-end test outstanding
Test deployment of changed terraform code outstanding

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have successfully run the E2E tests, and have included the links to the pipeline runs below
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have updated any relevant documentation to reflect my changes
- [ ] I have assigned a reviewer and messaged them

# Pipeline run links:

### CODE REVIEW

**Comment by browningjp-datatonic** [comment]
xgboost e2e pipeline failing because of another known issue (unrelated to this PR) https://github.com/GoogleCloudPlatform/vertex-pipelines-end-to-end-samples/pull/15

File: envs/prod/variables.auto.tfvars [hunk 1]
```diff
@@ -17,13 +17,107 @@ project_id = "my-prod-project"
...
-  training = {
-    name        = "training-pipeline-trigger",
-    region      = "europe-west4", # Must be the same as the App Engine region
-    description = "Trigger my training pipeline in Vertex",
-    schedule    = "0 0 * * 0",
-    time_zone   = "UTC",
-    # Relative path to payload JSON file for this environment
-    payload_file = "../../pipelines/xgboost/training/payloads/prod.json",
-  },
+  # Uncomment and amend as required
```
Comment (simonbutt-datatonic) [inline envs/prod/variables.auto.tfvars:20]: @browningjp-datatonic 
There must be a better approach than 100 lines of commented code. 
Would rather have open in cloud shell with tensorflow and xgboost as two example approaches

Comment (browningjp-datatonic) [reply envs/prod/variables.auto.tfvars:20]: Agree - but probably the refactor needed outside of the scope of this PR

File: pipelines/trigger/main.py [hunk 2]
```diff
@@ -191,25 +186,34 @@ def get_env() -> dict:
...
+        "--template_path", help="Path to the payload JSON file", type=str
+    )
+    parser.add_argument("--enable_caching", type=str, default=None)
+
+    # Get commandline args
+    args = parser.parse_args(args)
+
+    # If empty value for enable_caching provided on commandline default to None
+    if args.enable_caching == "":
+        args.enable_caching = None
```
Comment (simonbutt-datatonic) [inline pipelines/trigger/main.py:207]: Throw warning

Comment (browningjp-datatonic) [reply pipelines/trigger/main.py:207]: Why? Value can be `True` or `False` or `None`, not just True/False

**Review by simonbutt-datatonic — APPROVED** [review]
LGTM. Please fix the e2e test for next PR
